org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/DebugAdapter.java

```diff
@@ -782,4 +786,180 @@ private String convertDebuggerSourceToClient(String uri) {
     private Types.Message convertDebuggerMessageToClient(String message) {
         return new Types.Message(this.messageId.getAndIncrement(), message);
     }
+
+    private class VariableHandler {
+        private boolean showStaticVariables = true;
+        private IVariableFormatter variableFormatter;
+        private RecyclableObjectPool<ThreadReference, Object> objectPool;
+        private Map<String, Object> options;
+        
+        public VariableHandler(IVariableFormatter variableFormatter, boolean showStaticVariables,
+                               boolean hexFormat, boolean showQualified) {
```

Comment:
Why hexFormat is a parameter of the constructor? This should be an option passed to the formatters. This options should be set when processing each variable request.